Okta: Don't allow creating user_group Access Request in RO

kopiczko opened the pull request:
Issue https://github.com/gravitational/teleport.e/issues/6595

https://github.com/gravitational/teleport.e/pull/6630 should be merged first to not break .e tests.

changelog: UX: Forbid creating Access Requests to user_group resources when Okta bidirectional sync is disabled.

kopiczko committed:
Okta: Don't allow creating user_group Access Request in RO

smallinsky commented on lib/auth/okta/auth.go:
```diff
@@ -130,29 +125,53 @@ var (
+	})
+	var oktaAppServers []types.AppServer
+	if hasApps {
+		oktaAppServers, err = auth.GetApplicationServers(ctx, apidefaults.Namespace)
```
Instead of listing all application servers before performing any operation, would it be possible to call:
```
GetUnifiedResourcesByIDs(ctx, id)
```

for the `types.KindAppServer, types.KindApp:` case 

There is no need to load all app servers if the resourcesIDs are known.

kopiczko replied:
It can't be done. Added a comment here https://github.com/gravitational/teleport/compare/b66b164e9f3d95f4e8180062f8ee87124f970ebe..e4198668661ab5d886616f2d5d817731830bd1d7

kopiczko commented on lib/auth/okta/interfaces.go:
```diff
@@ -0,0 +1,37 @@
+	"github.com/gravitational/teleport/api/types"
+)
+
+type PluginGetter interface {
```
Do you mean unexported? It could be but then it's passed as an argument to a exported function. I know we use this package internally but I still like to treat those internal packages as a proper libraries. I'd expect to have public function argument types to be public. Otherwise the godoc is not visible. Also you may want to e.g. temporary store or pass around the implementation in the using package and then you'd have to create your own type.

greedy52 reviewed (approved):
`e` ref requires an update before merging?

kopiczko committed:
Switch getting app_server to ListResources

kopiczko commented:
@greedy52 yes `e` changes should be merge first to not break tests there.

I changed the implementation to suggested ListResources. There is a Label filter there and I added Limit:1 + predicate so should be efficient enough. We have a similar implementation in `AccessRequestReconciler` in `e`.

Thanks for the suggestion.
